src/lib/y2partitioner/dialogs/btrfs_subvolume.rb

```diff
@@ -0,0 +1,192 @@
+        def initialize(form, filesystem: nil)
+          @form = form
+          @filesystem = filesystem
+        end
+
+        def label
+          _("Path")
+        end
+
+        def store
+          form.path = value
+        end
+
+        def init
+          focus
+          self.value = form.path
+        end
+
+        # Validates the path
+        #
+        # Path cannot be empty
+        # Path must start by default subvolume path
+        # Path must be uniq
+        def validate
+          valid = true
+
+          focus
+
+          if value.empty?
+            Yast::Popup.Message(_("Empty subvolume path not allowed."))
+            valid = false
+          elsif !filesystem.nil?
+            fix_path
+            if exist_path?
+              Yast::Popup.Message(format(_("Subvolume name %s already exists."), value))
+              valid = false
+            end
+          end
+
+          valid
+        end
+
+      private
+
+        def focus
+          Yast::UI.SetFocus(Id(widget_id))
+        end
+
+        # Updates #value by prefixing path with default subvolume path if it is necessary
+        def fix_path
+          default_subvolume_path = default_path
+          prefix = default_subvolume_path + "/"
+
+          return value if value.start_with?(prefix)
+
+          message = format(
+            _("Only subvolume names starting with \"%s\" currently allowed!\n" \
+              "Automatically prepending \"%s\" to name of subvolume."), prefix, prefix
+          )
+          Yast::Popup.Message(message)
+
+          self.value = File.join(default_subvolume_path, value)
+        end
+
+        # If a default subvolume exists, its path is consider as default path.
+        # In case that the top subvolume is set as default one, the default path
+        # for default btrfs subvolumes is returned.
+        #
+        # @see Y2Storage::Filesystems::BlkFilesystem#default_btrfs_subvolume_path
+        #
+        # @return [String]
+        def default_path
+          default_subvolume = filesystem.default_btrfs_subvolume
+
+          if default_subvolume.nil? || default_subvolume.top_level?
+            filesystem.default_btrfs_subvolume_path
+          else
+            default_subvolume.path
+          end
+        end
+
+        # Checks if there is a subvolume with the entered path
+        def exist_path?
+          filesystem.btrfs_subvolumes.any? { |s| s.path == value }
+        end
+      end
+
+      # Input field to set the subvolume nocow attribute
+      class SubvolumeNocow < CWM::CheckBox
+        attr_reader :form
+        attr_reader :filesystem
+
+        # @param form [Dialogs::BtrfsSubvolume::Form]
+        # @param filesystem [Y2Storage::Filesystems::BlkFilesystem] a btrfs filesystem
+        def initialize(form, filesystem: nil)
+          @form = form
+          @filesystem = filesystem
+        end
+
+        def label
+          _("noCoW")
```

Comment:
I really suggest to write translator hint here. Consider that you are not IT expert and you have see such string, what it is? I suggest something like
```
# TRANSLATORS: noCoW is accronym to "not use Copy on Write" feature for Btrfs. If no suitable expression exists in your language, then better keep it as it is as it is expert value.
```